app/models/audit/event/authn.rb

```diff
@@ -81,8 +84,10 @@ def attempted_action
       def auth_stuctured_data
         { authenticator: @authenticator_name }.tap do |sd|
           sd[:service] = service_id if @service
+          sd[:user] = @success || Role[role_id: @role_id] ? @role_id : NOT_FOUND #Masking role if it doesn't exist to avoid audit pollution
```

Comment:
does this work? it doesn't look right. From what i see, the value of `sd[:user]` will be the value of `@success` if it's true, and not `Role[role_id: @role_id]`. 
am i wrong?

Reply:
why do we still check the `@success` value?

Reply:
@orenbm, from running the code it appears to be working. But I can add parenthesis around `@success || Role[role_id: @role_id]` to ensure that the ? applies for the entire phrase.
Why do we still check `@success`? to save a call to DB in case of success

Reply:
oh i see the logic now. yes, parentheses would help to understand that the condition is on `@success || Role[role_id: @role_id]`

Reply:
> Why do we still check @success? to save a call to DB in case of success
can you write this in a comment? it is not trivial.

I think that we have here enough logic to move this to a method and not use a one-liner (use `if` instead)